app/pages/MainPage/Main.js

```diff
@@ -54,66 +49,47 @@ class Main extends React.Component {
       dataSource: new ListView.DataSource({
         rowHasChanged: (row1, row2) => row1 !== row2
       }),
-      typeIds: [],
       typeList: {}
     };
   }
 
   componentDidMount() {
-    const { readActions } = this.props;
-    DeviceEventEmitter.addListener('changeCategory', (typeIds) => {
-      typeIds.forEach((typeId) => {
-        readActions.requestArticleList(false, true, typeId);
-        pages.push(1);
-      });
+    this.retrieveArticleLists(this.props.selectedCategoryIds);
+
+    store.get('typeList').then((typeList) => {
       this.setState({
-        typeIds
-      });
-    });
-    InteractionManager.runAfterInteractions(() => {
-      store.get('typeIds').then((typeIds) => {
-        if (!typeIds) {
-          return;
-        }
-        typeIds.forEach((typeId) => {
-          readActions.requestArticleList(false, true, typeId);
-          pages.push(1);
-        });
-        store.get('typeList').then(typeList =>
-          this.setState({
-            typeIds,
-            typeList
-          })
-        );
+        typeList
       });
     });
   }
 
   componentWillReceiveProps(nextProps) {
     const { read } = this.props;
+    this.retrieveArticleListsWhenNeeded(
```

Comment:
The problem is not stable and repetition.I have an idea:Why don't you try [FlatList](https://facebook.github.io/react-native/docs/flatlist.html) instead of ListView? I think it can solve this problem of the strange.